projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Subscribe modal popup feature
+ *
+ * @package automattic/jetpack-mu-wpcom
+ * @since 2.2.1
+ */
+
+/**
+ * Adds a Subscribe Modal template part that can be edited in the editor.
+ * This is later loaded in a pop up modal for Newsletter sites.
+ *
+ * @return void
+ */
+function wpcom_create_subscribe_template() {
+	if ( should_enable_subscriber_modal() ) {
+		$post = get_page_by_path( get_subscribe_template_slug(), OBJECT, 'wp_template_part' );
+
+		if ( ! $post ) {
+			$template = array(
+				'slug'         => get_subscribe_template_slug(),
+				'post_name'    => get_subscribe_template_slug(),
+				'post_title'   => get_subscribe_template_title(),
+				'post_content' => get_subscribe_template_content(),
+				'post_status'  => 'publish',
+				'post_author'  => 1,
+				'post_type'    => 'wp_template_part',
+				'scope'        => array(),
+				'tax_input'    => array(
+					'wp_theme' => get_option( 'stylesheet' ),
+				),
+			);
+			wp_insert_post( $template );
+		}
+	}
+}
+add_action( 'wp_loaded', 'wpcom_create_subscribe_template' );
+
+/**
+ * Adds modal with Subscribe Modal content.
+ *
+ * @return void
+ */
+function wpcom_add_subscribe_modal_to_frontend() {
+	if ( should_enable_subscriber_modal() && ! is_admin() ) {
+		$posts = get_posts(
+			array(
+				'post_type'   => 'wp_template_part',
+				'post_status' => 'publish',
+				'numberposts' => -1,
+			)
+		);
+
+		$subscribe_template = reset(
+			array_filter(
+				$posts,
+				function ( $post ) {
+					return $post->post_name === get_subscribe_template_slug();
+				}
+			)
+		);
+
+		$blocks = parse_blocks( $subscribe_template->post_content );
```

Comment:
Should we bail here on sites that won't have / support template parts?

Reply:
Good thought. I believe all FSE themes should support template parts? Assuming yes, I've added a check with `wp_is_block_theme()` here: 

https://github.com/Automattic/jetpack/pull/31393/files#diff-bcd4c897206a1f66102ef44f0e272985d67be37c91dfacbbd6458d46d66ac72eR188